README.md

```diff
@@ -1,2 +1,4 @@
 # lmp-el2go-auto-register
 A systemd oneshot helper to auto register a device using EdgeLock2GO
+
+- The files "default.env" and "root.crt" should be included with the user configurations.
```

Comment:
Where exactly the flles should be included? How should I populate them? is there a doc link you can add here?

Reply:
There is no docs page describing it. Should I add a PR in the docs about it?

Reply:
actually I'm wrong. The URL is: https://docs.foundries.io/latest/user-guide/el2g.html#creating-an-lmp-build-with-edgelock-2go

Reply:
These files should be included on the meta-sub layer or manualy installed on the target, `default.env` provides the default environment to run the tool:
https://github.com/foundriesio/lmp-el2go-auto-register/blob/302d47ee8e8daaa3febbfe3b4b27f80d16bb4aee/lmp-el2go-auto-register#L17-L22

The default environment is expeted to be provided by the systemd service and they should be installed to `/etc/default/lmp-el2go-auto-register `
https://github.com/foundriesio/lmp-el2go-auto-register/blob/302d47ee8e8daaa3febbfe3b4b27f80d16bb4aee/lmp-el2go-auto-register.service#L8

The `root.crt` should is expeted to be in /usr/share/lmp-el2go-auto-register/root.crt 
https://github.com/foundriesio/lmp-el2go-auto-register/blob/302d47ee8e8daaa3febbfe3b4b27f80d16bb4aee/lmp-el2go-auto-register#L243

Reply:
> Maybe this comment [1] should be integrated into [2], and the link [2] in integrated into the current README?

look like the better option but my knowledge about this tool is very limited and my comment was basically what is in the source code.

Reply:
@kprosise can you, please help here?
We are struggling to find how to handle this piece of doc. Do you know who can help review this?

Reply:
There are a few inaccuracies in @quaresmajose comment.
 * `default.env` must contain REPOID as described in the docs
```
Next, the recipe needs access to your “repo id”. The fioctl factories command will show your value. Put that in the file:

# recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register/default.env
REPOID=<YOUR ID FROM fioctl factories>
```
I believe other variables are optional.

`EnvironmentFile` is provided by the recipe in meta-lmp:
```
EnvironmentFile=-/etc/default/lmp-el2go-auto-register 
```
https://github.com/foundriesio/meta-lmp/blob/main/meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register.bb#L24

So IMHO there is nothing to be done by the user except for overwriting empty `default.env` from the recipe. Same goes for `root.crt`.

To conclude my take on the README text would be sth like this:
```
This script should be the part of LmP based distro.
To configure it properly user should at least provide values for REPOID and root.crt
as described in the Foundries documentation: https://docs.foundries.io/latest/user-guide/el2g.html#creating-an-lmp-build-with-edgelock-2go

Files default.env and root.crt should be part of meta-subscriber-overrides layer. 
They should replace empty files from meta-lmp layer.
Without these files generated images 
will not contain FoundriesFactory specific values 
and EdgeLock2GO registration will not work.
```

Reply:
Thanks @mwasilew added your text to the README

Reply:
The README text is great.
But we still need a User Guide on how to change the files for the factory
And if the files "should be part" of meta-sub, maybe they have to go to the template.  All the architectures need those files, or this is only for some arch?

If I create a factory without those files, what happens to my registration process?

(I'm sorry if I'm making too many questions you already answered or are working to fulfill, it's because I don't have the full context of this change so I'm trying to review based on zero context)

Reply:
I think it's all described in the doc I linked.

> All the architectures need those files, or this is only for some arch?

All. This is not arch specific. I'm not sure there is se050 for x86 though.

> If I create a factory without those files, what happens to my registration process?

EL2GO autoregistration will fail. Manual registration will work. This patch is a result of our autoregistration test failing btw.

Reply:
I found it now. It's 
![image](https://github.com/foundriesio/lmp-el2go-auto-register/assets/1568936/4f654efc-897d-41f6-87a6-9aaccc652e82)
sorry for the noise